File: sample-objc/ARGEARSample/ARGearHelpers/ARGPreferences/ARGPreferences.m
//
//  ARGPreferences.m
//  ARGEARSample
//
//  Created by Jihye on 13/11/2019.
//  Copyright © 2019 Seerslab. All rights reserved.
//

#import "ARGPreferences.h"

@implementation ARGPreferences

- (id)init {
    self = [super init];
    if (self) {
        _showLandmark = [self loadBoolValue:kARGPreferencesShowLandmark];
        _videoBitrate = [self loadIntegerValue:kARGPreferencesVideoBitrate];
    }
    
    return self;
}

// load
- (BOOL)loadBoolValue:(NSString *)key {
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    return [userDefaults boolForKey:key];
}

- (NSInteger)loadIntegerValue:(NSString *)key {
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    return [userDefaults integerForKey:key];
}

// store
- (void)storeBoolValue:(BOOL)value key:(NSString *)key {
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setBool:value forKey:key];
}

- (void)storeIntegerValue:(NSInteger)value key:(NSString *)key {
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setInteger:value forKey:key];
}

- (void)setShowLandmark:(BOOL)showLandmark {
    
    _showLandmark = showLandmark;
    
    [self storeBoolValue:showLandmark key:kARGPreferencesShowLandmark];
}

- (void)setVideoBitrate:(NSInteger)videoBitrate {
    
    _videoBitrate = videoBitrate;
    
    [self storeIntegerValue:videoBitrate key:kARGPreferencesVideoBitrate];
}

@end

File: sample-swift/ARGearSample/ARGearHelpers/ARGPreferences/ARGPreferences.swift
//
//  ARGPreferences.swift
//  ARGearSample
//
//  Created by Jihye on 2020/01/29.
//  Copyright © 2020 Seerslab. All rights reserved.
//

import Foundation


class ARGPreferences {
    
    let kARGPreferencesShowLandmark = "kARGPreferencesShowLandmark"
    let kARGPreferencesVideoBitrate = "kARGPreferencesVideoBitrate"

    var showLandmark: Bool = false
    var videoBitrate: NSInteger = 0
    
    init() {
        self.showLandmark = self.loadBoolValue(kARGPreferencesShowLandmark)
        self.videoBitrate = self.loadIntegerValue(kARGPreferencesVideoBitrate)
    }

    // load
    func loadBoolValue(_ key: String) -> Bool {
        
        let userDefaults = UserDefaults.standard
        return userDefaults.bool(forKey: key)
    }

    func loadIntegerValue(_ key: String) -> NSInteger {
        
        let userDefaults = UserDefaults.standard
        return userDefaults.integer(forKey: key)
    }

    // store
    func storeBoolValue(_ value: Bool, key: String) {
        
        let userDefaults = UserDefaults.standard
        
        userDefaults.set(value, forKey: key)
        userDefaults.synchronize()
    }

    func storeIntegerValue(_ value: NSInteger, key: String) {
        
        let userDefaults = UserDefaults.standard
        
        userDefaults.set(value, forKey: key)
        userDefaults.synchronize()
    }

    func setShowLandmark(_ showLandmark: Bool) {
        
        self.showLandmark = showLandmark
        self.storeBoolValue(showLandmark, key: kARGPreferencesShowLandmark)
    }

    func setVideoBitrate(_ videoBitrate: NSInteger) {
        
        self.videoBitrate = videoBitrate
        self.storeIntegerValue(videoBitrate, key: kARGPreferencesVideoBitrate)
    }
}
